File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max_price = 0;
        int x = prices[0];
        for(int i=1;i<prices.length;i++){
            if(prices[i]<x){
                x = prices[i];
            }else if(prices[i] - x > max_price){max_price = prices[i] - x;}
        }
        return max_price;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        diff = 0
        i = 1

        while i < len(prices):
            if prices[i] < buy:
                buy = prices[i]
            else:
                diff = max(diff, prices[i] - buy)

            i += 1

        return diff

